docs/Secure-Coding-Guide-for-Python/CWE-703/CWE-755/README.md

```diff
@@ -0,0 +1,194 @@
+
+*[noncompliant01.py](noncompliant01.py):*
+
+```py
+""" Non-compliant Code Example """
+ 
+import os
+import uuid
+ 
+ 
+def read_file(file):
+    """Function for opening a file and reading it's content."""
+    fd = os.open(file, os.O_RDONLY)
+    content = os.read(fd)
+    return content.decode()
+ 
+ 
+#####################
+# exploiting above code example
+#####################
+# Attempting to read a random non-existent file
+read_file(f"{uuid.uuid4()}.txt")
+```
+
+## Compliant Solution (try/except blocks)
+
+The file opening and reading should be surrounded by the `try/except` block. This way, we can catch the generic `OSError` and handle the error differently depending on its cause (such as the file not existing or it being a directory instead).
+
+*[compliant01.py](compliant01.py):*
+
+```py
+""" Compliant Code Example """
+  
+import os
+import uuid
+  
+  
+def read_file(file):
+    """Function for opening a file and reading its content."""
+    try:
+        fd = os.open(file, os.O_RDONLY)
+        try:
+            content = os.read(fd, 1024)
+        finally:
+            os.close(fd)
+        return content.decode()
+    except OSError as e:
+        if not os.path.exists(file):
+            print(f"File not found: {file}")
+        elif os.path.isdir(file):
+            print(f"Is a directory: {file}")
+        else:
+            print(f"An error occurred: {e}")
+  
+  
+#####################
+# exploiting above code example
+#####################
+# Attempting to read a random non-existent file
+read_file(f"{uuid.uuid4()}.txt")
+```
+
+## Non-Compliant Code Example (pathlib.Path.unlink())
+
+The `pathlib` module also provides functions for opening and reading files. The `Path.read_text()` method attempts to read the content of the file represented by the `Path` object. If the file does not exist, it will raise a `FileNotFoundError`. In this code example, this exception is expected when attempting to read a randomly generated non-existent file.
+
+*[noncompliant02.py](noncompliant02.py):*
+
+```py
+""" Non-compliant Code Example """
+ 
+import uuid
+from pathlib import Path
+ 
+ 
+def read_file(file):
+    """Function for opening a file and reading its content."""
+    path = Path(file)
+    content = path.read_text(encoding="utf-8")
+    return content
+ 
+ 
+#####################
+# exploiting above code example
+#####################
+# Attempting to read a random non-existent file
+read_file(f"{uuid.uuid4()}.txt")
+```
+
+The `pathlib.Path.unlink()` function has an optional parameter `missing_ok` that will suppress the `FileNotFoundError on file deletion`, if the parameter's value is `True` . However, without proper handling, using this parameter will cause the script to fail silently.
```

Comment:
I believe that was a typo on the original page. It should be: `FileNotFoundError` on file deletion